### DISCUSSION TITLE
Custom Component used in formik causes resetForm issue

### DISCUSSION THREAD

**Discussion by ron-ts** [jaredpalmer/formik#3929] [General]
Hi,

I've been building a custom component wrapping the LexicalComposer component which can be found [online](https://lexical.dev/docs/intro).

I'm using this lexical composer in-order to allow users to write description in some richText format.

I am using formik and using <Field as={myCustomLexicalComposer}/> does not serve me as the values from formik does not work as I expect because (i guess) the LexicalComposer is not based on html <input/>.

To make it work I've handled it manually by listening to onChange events and update the field myself

i.e something like:

onChange={(event) => {setFieldValue("description", event.target.value)}}

One of the functionalities on this page is when the user decides not to save the changes, hitting 'cancel' reverts the form to the initial form values using 'resetForm()' function.

This does not work for me anymore for some reasons: 

1. I suspect that the non - input component I'm using is not auto catched by formik and therefore causes problem
2. For some reason, even if I'm specifically handling 'description' field after resetForm() function the change does not take effect in the formik's values (such as setFieldValue("description", initialValues.description)) i.e.:
```
function handleCancel {
  resetForm()
  setFieldValue("description", initialValue.description)
}
```
3. If I wait long enough the values are being updated but it does not affect the component itself (async action)

By async action I mean that the definition of my 'cancel' button is an onClick that I register an async function, tried to remove the async and it did not solve the issue.

I'm using formik v2.+.+

Would appreciate any assistance 🥲

**Comment by ghost** [comment]
Hi! Can you provide us some code by any chance?
